model/src/application/interaction/mod.rs

```diff
@@ -54,13 +54,25 @@ pub enum Interaction {
 }
 
 impl Interaction {
+    /// ID of the user that invoked the interaction.
+    pub const fn author_id(&self) -> Option<Id<UserMarker>> {
+        match self {
+            Interaction::Ping(_) => None,
+            Interaction::ApplicationCommand(command) => command.author_id(),
+            Interaction::ApplicationCommandAutocomplete(command) => command.author_id(),
+            Interaction::MessageComponent(component) => component.author_id(),
+            Interaction::ModalSubmit(modal) => modal.author_id(),
+        }
+    }
+
+    /// Return the guild ID the interaction was run in.
```

Comment:
We can simplify this and use the same term as above, "invoked" -- I think we use it elsewhere as well, good call on that.
```suggestion
    /// ID of the guild the interaction was invoked in.
```